tensornetwork/backends/jax/jitted_functions.py

```diff
@@ -567,54 +586,400 @@ def gmres_m(A_mv: Callable, A_args: Sequence,
     n_iter (int)    : Number of iterations at termination.
     converged (bool): Whether the desired tolerance was achieved.
     """
+    num_krylov_vectors = min(num_krylov_vectors, b.size)
     x = x0
-    converged = False
-    r, beta = gmres_residual(A_mv, A_args, b, x)
     b_norm = jnp.linalg.norm(b)
+    tol = max(tol*b_norm, atol)
     for n_iter in range(maxiter):
-      # pylint: disable=too-many-function-args
-      x = gmres(A_mv, A_args, num_krylov_vectors, x, r, beta, tol)
-      r, beta = gmres_residual(A_mv, A_args, b, x)
-      if beta <= max(tol*b_norm, atol):
-        converged = True
+      done, beta, x = gmres(A_mv, A_args, b, x, num_krylov_vectors, x0, tol,
+                            b_norm)
+      if done:
         break
-    return (x, beta, n_iter, converged)
+    return (x, beta, n_iter, done)
+
+
+  def gmres(A_mv: Callable, A_args: Sequence, b: jax.ShapedArray,
+            x: jax.ShapedArray, num_krylov_vectors: int, x0: jax.ShapedArray,
+            tol: float, b_norm: float) -> Tuple[bool, float, jax.ShapedArray]:
+    """
+    A single restart of GMRES.
+
+    ARGS
+    ----
+    A_mv     : A function `v0 = A_mv(v, *A_args)` where `v0` and
+               `v` have the same shape.
+    A_args   : A list of positional arguments to A_mv.
+    b        : The `b` in `A @ x = b`.
+    x        : Initial guess solution.
+    tol      : Solution tolerance to achieve,
+    num_krylov_vectors : Size of the Krylov space to build.
+
+    RETURNS
+    -------
+    done     : Whether convergence was achieved.
+    beta     : Magnitude of residual (i.e. the error estimate).
+    x        : The approximate solution.
```

Comment:
Please use argument formatting to be consistent with the rest of the library.

Reply:
fair enough. Is it just ARGS, RESULTS -> Args: Results:?

Reply:
Yeah, and the spacing. Take a look at any other function to see an example. I might add docs check in the future.

Reply:
Ok, should be fixed.